requests/sessions.py

```diff
@@ -289,7 +289,9 @@ def rebuild_proxies(self, prepared_request, proxies):
         new_proxies = proxies.copy()
         no_proxy = proxies.get('no_proxy')
 
-        bypass_proxy = should_bypass_proxies(url, no_proxy=no_proxy)
+        bypass_proxy = False
+        if self.trust_env:
```

Comment:
@omermizr  / @sigmavirus24  - this change fixes the performance regression of 5891 but does not fix 5888. IMHO fixing 5888 is something that you might do by checking if a proxy has been selected as if one has not been selected you likely would not want to leak proxy credentials. Of course it might be easier to suggest that we do not explicitly support setting `'Proxy-Authorization'` as a header and that users must specify proxy credentials in the proxy url.